src/lib.rs

```diff
@@ -242,6 +242,20 @@ pub fn write_ivf_frame(output_file: &mut Write, pts: u64, data: &[u8]) {
     output_file.write(data).unwrap();
 }
 
+fn write_frame_size(fi: &FrameInvariants, uch: &mut BitWriter<BE>) -> Result<(), std::io::Error> {
```

Comment:
You might put in a trait so you have `uch.write_frame_size(fi)`;

Reply:
That would be interesting, it would be a bigger refactor though

Reply:
I can give it a go once this is merged then.

Reply:
Sounds good